File: z-clients/javascript/src/models/sinkDeleteOut.ts
// this file is @generated

export interface SinkDeleteOut {
    topic: string;
    consumerGroup: string;
    success: boolean;
}

export const SinkDeleteOutSerializer = {
    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _fromJsonObject(object: any): SinkDeleteOut {
        return {
            topic: object['topic'],
            consumerGroup: object['consumer_group'],
            success: object['success'],
        };
    },

    // biome-ignore lint/suspicious/noExplicitAny: intentional any
    _toJsonObject(self: SinkDeleteOut): any {
        return {
            'topic': self.topic,
            'consumer_group': self.consumerGroup,
            'success': self.success,
        };
    }
}

File: z-clients/rust/src/models/sink_delete_out.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Deserialize, Serialize)]
pub struct SinkDeleteOut {
    pub topic: String,

    pub consumer_group: String,

    pub success: bool,
}

impl SinkDeleteOut {
    pub fn new(topic: String, consumer_group: String, success: bool) -> Self {
        Self {
            topic,
            consumer_group,
            success,
        }
    }
}
